src/zenml/client.py

```diff
@@ -774,6 +775,13 @@ def register_stack_component(
 
         # Update the configuration in the model
         component.configuration = configuration
+        if (
+            self.zen_store.is_local_store()
+            and not StackComponent.from_model(component).is_local
+        ):
+            logger.warn(
+                "You are registering a non-local stack component while using a local ZenServer. This may lead to unexpected behavior."
```

Comment:
Shouldn't we be more forcefull? A remote orchestrator with a local zensotre will certainly fail, so we should maybe even strongly discourage this and explain why this might not work

Reply:
I updated the message to convey more information. I would still not make this an error, as history showed people have all sorts of deployment use-cases, some of which are on-prem or even on the same machine, but we have no easy way of detecting those.